### PR TITLE
Added sched_getaffinity and sched_setaffinity for Android; moved thes…

### PR DESCRIPTION
**Pull request by raphaelcohn**
…e up from Linux.

Added pthread_gettid_np for Android

Added pthread_setaffinity_np and pthread_getaffinity_np for Linux

### CODE REVIEW

**Comment by alexcrichton** [comment]
May need `cpu_set_t` definitions in a few places?

**Comment by raphaelcohn** [comment]
Yep.

File: src/unix/notbsd/linux/mod.rs [hunk 1]
```diff
@@ -627,6 +621,10 @@ extern {
                              remote_iov: *const ::iovec,
                              riovcnt: ::c_ulong,
                              flags: ::c_ulong) -> isize;
+    pub fn pthread_setaffinity_np(thread: ::pthread_t, cpusetsize: ::size_t,
+                             cpuset: *const ::cpu_set_t) -> ::c_int;
```
Comment (alexcrichton) [inline src/unix/notbsd/linux/mod.rs]: Looks like some alignment is off here?

File: src/unix/notbsd/android/mod.rs [hunk 2]
```diff
@@ -87,6 +88,13 @@ s! {
         pub l_len: ::off_t,
         pub l_pid: ::pid_t,
     }
+
+    pub struct cpu_set_t {
+        #[cfg(target_pointer_width = "64")]
+        __bits: [__CPU_BITTYPE; 16],
+        #[cfg(target_pointer_width = "32")]
+        __bits: [__CPU_BITTYPE; 1],
+    }
```
Comment (alexcrichton) [inline src/unix/notbsd/android/mod.rs]: This was included in another PR, right?

Comment (raphaelcohn) [inline src/unix/notbsd/android/mod.rs]: Possibly... I had a short span of time over the weekend and strongly suspect I merged a change into the wrong PR...

File: src/unix/notbsd/linux/mod.rs (hunk 1 above)
Comment (raphaelcohn) [inline src/unix/notbsd/linux/mod.rs]: Will take a look

**Comment by alexcrichton** [comment]
Thanks! Wanna squash these down and rebase? Other than that r=me

**Comment by alexcrichton** [comment]
Closing due to inactivity, but feel free to resubmit with a rebase!

**Pull request closed without merging**